### DISCUSSION TITLE
pki --scep function implementation, there are currently supported versions

### DISCUSSION THREAD

**Discussion by XYQ202** [strongswan/strongswan#1305] [Q&A]
Hello Teams,

        When I read your document, I saw the function of pki --scep. I saw that the change function is mainly to register the certificate with SCEP and update the certificate. Is this function?
        Because I have already implemented the function of CMPv2, which is also the certificate registration and renewal, I still want to implement the function of SCEP, pki --scep will also have the same function as CMP IR or KUR, will it send a request to the SCEP server?
        When will I be able to use this new feature, also, do you know where there is a SCEP server?

**Comment by strongX509** [comment]
**`pki --scep`** implements the SCEP RFC 8894 and can request an X.509 certificate from a CA based on a secret challenge password or via manual authentication. Certificate renewal based on the old certificate is also supported. For details have a look at our documentation:

https://docs.strongswan.org/docs/5.9/pki/pkiScep.html

**Comment by strongX509** [comment]
You can test the **`pki --scep`** functionality by downloading the [strongswan-5.9.8dr3 tarball](https://download.strongswan.org/strongswan-5.9.8dr3.tar.bz2). I have been testing with the [openxpki docker image](https://github.com/openxpki/openxpki-docker) which has a SCEP server.

**Reply by XYQ202** [reply]
> You can test the **`pki --scep`** functionality by downloading the [strongswan-5.9.8dr3 tarball](https://download.strongswan.org/strongswan-5.9.8dr3.tar.bz2). I have been testing with the [openxpki docker image](https://github.com/openxpki/openxpki-docker) which has a SCEP server.

Thanks for your reply, I'll try it myself. At the same time, I installed EJBCA, which is an open source CA server that supports CMP, SCEP, etc., but I configured SCEP Alias on this server, but the CA certificate obtained through the ipsec pki --scepca command failed, and the error is as follows:
![企业微信截图_16633161523041](https://user-images.githubusercontent.com/97773232/190591604-b54159b2-3d2b-4265-90af-066b2ff07bd1.png)

**Reply by strongX509** [reply]
The **`pki`** tool needs the **`curl`** plugin in order to be able to make http requests to the SCEP server.  Compile the strongswan tarball with the  **`--enable-curl`** ./configure option. Also the libcurl header files are needed for the compilation

**Reply by XYQ202** [reply]
> The **`pki`** tool needs the **`curl`** plugin in order to be able to make http requests to the SCEP server. Compile the strongswan tarball with the **`--enable-curl`** ./configure option. Also the libcurl header files are needed for the compilation

![Uploading 企业微信截图_16635508203838.png…]()
What's the problem, has your pki --scepca function been connected to the real SCEP server?

**Reply by XYQ202** [reply]
> The **`pki`** tool needs the **`curl`** plugin in order to be able to make http requests to the SCEP server. Compile the strongswan tarball with the **`--enable-curl`** ./configure option. Also the libcurl header files are needed for the compilation

![企业微信截图_16635508203838](https://user-images.githubusercontent.com/97773232/190937387-23ad6efc-e1ed-49bc-a664-656c92c64b96.png)

**Reply by strongX509** [reply]
You **must not** use `HTTPS` with the `SCEP` protocol since it is in itself already protected: Thus the URL should be
````
--url http://192.168.32.150:8080/scep
````
I've been using the URL above with **`openxpki`**. Depending on the SCEP server and the TCP server port it is listening on, the URL might have to be modified, e.g. The [SCEP RFC 8894](https://datatracker.ietf.org/doc/html/rfc8894#section-4.1) defines
````
--url http://scepserver/cgi-bin/pkiclient.exe 
````

**Reply by XYQ202** [reply]
> 

![企业微信截图_16635782274168](https://user-images.githubusercontent.com/97773232/190984228-b99006a4-20b8-4490-8193-5688f8b91909.png)
Thank you for your reply and know, I have succeeded, thank you!
